src/crimson/os/seastore/transaction.h

```diff
@@ -130,16 +133,35 @@ class Transaction {
   ///< if != NULL_SEG_ID, release this segment after completion
   segment_id_t to_release = NULL_SEG_ID;
 
-  Transaction(bool weak) : weak(weak) {}
+  Transaction(
+    OrderingHandle &&handle,
+    bool weak
+  ) : handle(std::move(handle)), weak(weak) {}
+
+public:
+  ~Transaction() {
+    for (auto i = write_set.begin();
+	 i != write_set.end();) {
+      i->state = CachedExtent::extent_state_t::INVALID;
+      write_set.erase(*i++);
+    }
+  }
 };
 using TransactionRef = Transaction::Ref;
 
 inline TransactionRef make_transaction() {
-  return std::unique_ptr<Transaction>(new Transaction(false));
+  return std::unique_ptr<Transaction>(
+    new Transaction(
+      get_dummy_ordering_handle(),
+      false
+    ));
 }
 
 inline TransactionRef make_weak_transaction() {
-  return std::unique_ptr<Transaction>(new Transaction(true));
+  return std::unique_ptr<Transaction>(
+    new Transaction(
+      get_dummy_ordering_handle(),
```

Comment:
True, but I expect to want to use the embedded op at some point in the future to do tracing.  The intention is that the osd will pass in the op being processed instead of the dummy op used here.